File: 1-two-sum/2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int a = nums[i];
            int b = target - a;
            if (map.containsKey(b)) return new int[] {map.get(b), i};
            map.put(a, i);
        }
        return new int[]{};
    }
}

File: 1-two-sum/2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, a in enumerate(nums):
            b = target - a
            if b in map:
                return [map[b], i]
            map[a] = i
        return []
